src/ng/directive/ngOptions.js

```diff
@@ -489,6 +495,9 @@ var ngOptionsDirective = ['$compile', '$parse', function($compile, $parse) {
 
         selectCtrl.readValue = function readNgOptionsValue() {
 
+          // Allow a default value to be substituted for any options with empty values
+          if (selectElement.val() === '') return $parse(attr.ngEmptyValue)(scope);
```

Comment:
I'm a little lost on which test to update for this - it appears that this test...
`    it('should update model to null on change', function() {
      createSingleSelect(true);

      scope.$apply(function() {
        scope.values = [{name: 'A'}, {name: 'B'}];
        scope.selected = scope.values[0];
      });

      element.val('');
      browserTrigger(element, 'change');
      expect(scope.selected).toEqual(null);
    });`
...already checks that the model value is null on empty values in a no ng-empty-value situation? It does so implicitly, though.